### ISSUE TITLE
Retrieved data is truncated

### ISSUE THREAD

**Issue by obones** [boblemaire/esp32HTTPrequest#7]
I'm using `est32HTTPrequest` to `GET` a JSON document at a given URL and am expecting to receive 1386 bytes but for some reason `responseText` only contains 1381 characters

```
Debug(26593): setDebug(on) version 1.1.15
Debug(26596): open(GET, http://someserver/rest/item)
Debug(  0): _parseURL() HTTP://someserver/rest/item
Debug(  9): send()
Debug(  9): _send() 0
Debug(221): client connected event
Debug(221): _setReadyState(1)
Debug(224): headers sent event
Debug(431): header received event Content-Type:application/json
Debug(431): header received event Date:Sat, 20 Nov 2021 14:21:21 GMT
Debug(431): header received event Transfer-Encoding:chunked
Debug(437): header received event Server:Jetty(9.4.40.v20210413)
Debug(442): on-data event, len=358
Debug(445): _setReadyState(2)
Debug(448): _onData handler {"name":"uicompo... (358)
Debug(453): Response is chunked.
Debug(456): _setReadyState(3)
Debug(459): on-data event, len=512
Debug(461): _setReadyState(2)
Debug(464): _onData handler 25,8 °C]","icon... (512)
Debug(469): _setReadyState(3)
Debug(473): on-data event, len=512
Debug(474): _setReadyState(2)
Debug(477): _onData handler ","type":"Roller... (512)
Debug(482): _setReadyState(3)
Debug(485): disconnect event
Debug(489): client finish event
Debug(491): _setReadyState(4)
Debug(499): setDebug(off) version 1.1.15
```

If I add the lengths in the `_onData` handler, I get 1382 which is still 4 short of what I am expecting.
I was using `asyncHTTPrequest` before and did not have this truncation issue with it, but I know it's based on totally different library. To be frank, I'm trying `esp32HTTPrequest` out because of the SSL support (not yet used) and because from times to times, `asyncHTTPrequest`  would actually block on the call to `send()`.

Do you have any suggestions as to what I should try? Is there an option that I have missed?
I have already spied on the network connection with Wireshark and I do see the expected 1386 bytes coming through.

**Comment by obones** [comment]
If I set the `buffer_size` property of the `esp_http_client_config_t` to 1024, I now get two calls to `_onData handler`:

```
Debug(12072): _onData handler {"name":"uicompo... (870)
Debug(12089): _onData handler ","type":"Roller... (516)
```

The sum is now fine (870 + 516 = 1386), but when I call `responseText()`, I only get 1385 characters which still gives me a broken JSON obviously.

Could this come from the chunked output?

**Comment by obones** [comment]
> Could this come from the chunked output?

Actually, yes, that's because there is a miscalculation of `_contentLength` as shown in PR #8

**Comment by boblemaire** [comment]
Merged PR#8 to fix.  Version will be 1.1.17

**Comment by obones** [comment]
Actually, the PR only solves part of the issue. I mean, it allows a complete download when setting `buffer_size` to 1024
But if I leave it to its default value, then I get three calls to the `onData` event but the sum of received `len` is 4 bytes short of what I expect.
